companion/discord_bot/bot.py

```
@@ -0,0 +1,435 @@
+
+        # Initialize Redis client for distributed rate limiting
+        try:
+            self.redis_client = Redis.from_url(
+                self.redis_url,
+                encoding="utf-8",
+                decode_responses=True
+            )
+            # Test connection
+            await self.redis_client.ping()
+            self.logger.info(f"Redis client initialized successfully at {self.redis_url}")
+        except Exception as e:
```

Ruff (0.13.2)

Do not catch blind exception: `Exception`

(BLE001)